### ISSUE TITLE
Recognize Bot's messages

### ISSUE THREAD

**Issue by fgr1986** [sohlex/botkit-rasa#2]
Hi!

In order to not sent to rasa the bot reply, this could be added just after

```
 if (!message.text || message.is_echo) {
        next()
        return
}
```

we can simply identify whether the message comes from a bot:

```
if( message.bot_id != null ){
        console.log('\t\t[own_debug]  bot response, not processing')
        next()
        return
}
```
Hope it helps. Regards

**Comment by sohlex** [comment]
Hi!
Can you help me replicate this issue? For which platform are you developing the bot? Slack?

**Comment by fgr1986** [comment]
Sure.
When accomplishing the mini-tutorial, https://rasa-nlu.readthedocs.io/en/latest/tutorial.html
After say 'hello' to the bot (you send this message to rasa through the request/hears functions) the bot reply is 'Hello!!!, tell me how can I help you'. This message is feedback from the slack web to the botkit-rasa handler, with a message like:

```
message { type: 'message',
  user: 'YYYY',
  text: 'Hello!!!, tell me how can I help you',
  bot_id: 'YYYY',
  team: 'YYYY',
  channel: 'YYYY',
  event_ts: 'YYYY',
  ts: 'YYYY' }
```
If not filtered, this message will be again processed by rasa.
As this message has the `bot_id` defined, we can filter it using

```
if( message.bot_id != null ){
        console.log('\t\t[own_debug]  bot response, not processing')
        next()
        return
      }
```
Regards

**Comment by sohlex** [comment]
Got it, thanks! 👍 
Regarding this topic I see that there's also an open issue on Botkit: https://github.com/howdyai/botkit/pull/694

I've updated the plugin, now we ignore all the messages coming from other bots (technically all the message with a defined `bot_id`).
I'll (try to) keep an eye on the issue that I referenced before, maybe there will be a cleaner solution.

Thank you very much for the detailed report!
